From 746bf5f78376e2e7b3e8bc843dfa1e4a3fa6a8e4 Mon Sep 17 00:00:00 2001
From: Andrew Gregory <andrew.gregory.8@gmail.com>
Date: Mon, 11 Feb 2013 22:13:27 -0500
Subject: [PATCH] consolefont: use setfont to save the current font

Using setfont directly to save the font prevents breakage
when a distro stores consolefonts in a location other than
/usr/share/consolefonts such as Arch which stores them in
/usr/share/kbd/consolefonts

Signed-off-by: Andrew Gregory <andrew.gregory.8@gmail.com>
---
 init.d/consolefont.in     | 8 ++------
 sh/init-early.sh.Linux.in | 3 +--
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/init.d/consolefont.in b/init.d/consolefont.in
index 66b780d..0e91d13 100644
--- a/init.d/consolefont.in
+++ b/init.d/consolefont.in
@@ -53,14 +53,10 @@ start()
 	done
 	eend $retval
 
-	# Store the last font so we can use it ASAP on boot
+	# Store the font so we can use it ASAP on boot
 	if [ $retval -eq 0 ] && checkpath -W "$RC_LIBEXECDIR"; then
 		mkdir -p "$RC_LIBEXECDIR"/console
-		for font in /usr/share/consolefonts/"$consolefont".*; do
-			:
-		done
-		cp "$font" "$RC_LIBEXECDIR"/console
-		echo "${font##*/}" >"$RC_LIBEXECDIR"/console/font
+		setfont -O "$RC_LIBEXECDIR"/console/font
 	fi
 
 	return $retval
diff --git a/sh/init-early.sh.Linux.in b/sh/init-early.sh.Linux.in
index fb3860a..f9d2388 100644
--- a/sh/init-early.sh.Linux.in
+++ b/sh/init-early.sh.Linux.in
@@ -31,9 +31,8 @@ if service_present "$RC_DEFAULTLEVEL" consolefont ||
    service_present "$RC_BOOTLEVEL" consolefont; then
 	printf "\033%s" "$termencoding" >"$CONSOLE" 2>/dev/null
 	if [ -r "$RC_LIBEXECDIR"/console/font -a -x /usr/bin/setfont ]; then
-		font="$(cat "$RC_LIBEXECDIR"/console/font)"
 		[ -c "$CONSOLE" ] && cons="-C $CONSOLE"
-		setfont $cons "$RC_LIBEXECDIR"/console/"$font" 2>/dev/null
+		setfont $cons "$RC_LIBEXECDIR"/console/font 2>/dev/null
 	fi
 fi
 
-- 
1.8.3

